Update the CA override guide with new features and commands (#68402)
gravitational/teleport: docs/pages/zero-trust-access/management/security/ca-overrides.mdx

$ tctl auth create-override-csr --type=windows --subject='OU=My Organization Unit,CN=My Windows CA'
```

> ### Prepare CSRs for multi-Auth clusters with per-instance HSM keys

Reviewer: Would just "Prepare CSRs" work? Then we can clarify the actual scope of the section in the first paragraph. For me, "Prepare CSRs" seems distinct enough from the other headings to indicate its purpose within the high-level flow.

Author: "Prepare CSRs" is essentially the title of the header that precedes it.

I did here exactly what we did for the old "Create certificate overrides on multi-Auth clusters". Let me know if that's OK.

Author: 1517d4c5112

### db_client

> <details>

Author: My thinking for adding these is that, the more CAs we support, the more text folks have to scroll before they get to the parts they are looking for.

Happy to revert if you'd rather just have the text.

Reviewer: Right now, I'm finding this section of the rendered docs a little difficult to read:

<img width="639" height="271" alt="Image" src="https://github.com/user-attachments/assets/e04e5358-4001-41bb-99ad-d0d6ce44c3ba" />

Once we document more CAs here, I think it makes more sense to refactor, but for now, the content we're including in details boxes isn't very long.

Author: Reverted.

After issuing certificates and configuring downstream trust, create the
certificate overrides.

```code
$ tctl auth create-override --type=windows cert.pem
```

### Disabled overrides

You may create a disabled override and enable it later. A disabled override
works as if it didn't exist, leaving signing to the self-signed CA certificate.

This is useful for environments with staged rollouts, so overrides may be
created in a disabled state until downstream services are ready to accept them.
Set `disabled` to `false` when you are ready to enable the override.

Create the disabled override:

```code
$ tctl auth create-override --type=windows cert.pem --disabled
```

Enable it:

```code
$ tctl auth update-override --type=windows --public-key=<Var name="public-key" /> --set-disabled=false
```

You can find the public key of the target override by running
`tctl auth pub-key-hash --cert=cert.pem` or reading it from
`tctl get ca_overrides/windows`. Alternatively, you may edit directly using

> `tctl edit ca_overrides/windows`.

Reviewer: I would separate each of these commands into a `code` snippet for easier copying and pasting.

Author: Done.

Author: df47e16b98b

> ### Create certificate overrides on multi-Auth clusters with per-instance HSM keys

Reviewer: ```suggestion
### Multi-Auth Service clusters with per-instance HSM keys
```
Removing "certificate overrides" since that's implied by the topic of the guide. Using the extra space to expand "Multi-Auth" to "Multi-Auth Service", which is more consistent with the language used throughout the documentation.

Author: Done.

> In multi-Auth clusters with per-instance HSM keys, each Auth Service instance

Reviewer: ```suggestion
In clusters with multiple Auth Service instances and a separate HSM key per instance, each Auth Service instance
```
Expanding "multi-Auth" since that's not a term we use throughout the docs .

Author: Done.

After issuing certificates and configuring downstream trust, create the
certificate overrides.

```code

> $ tctl auth create-override --type=windows cert.pem

Author: 6b2c5b81308

After issuing certificates and configuring downstream trust, create the
certificate overrides.

```code
$ tctl auth create-override --type=windows cert.pem
```

> <Admonition type="tip">

Reviewer: This Admonition is quite long, so I would put it in its own section. For detailed instructions that aren't required to complete the body of a how-to guide, we often put them in an H2 at the end of the guide. One example of this is the Connecting Apps guide: https://goteleport.com/docs/enroll-resources/application-access/protect-apps/connecting-apps/

Author: I moved it to an H3 where it sits, similar to the "multi-Auth" section. Happy to move it to the end if you prefer.

created in a disabled state until downstream services are ready to accept them.
Set `disabled` to `false` when you are ready to enable the override.

Create the disabled override:

```code
$ tctl auth create-override --type=windows cert.pem --disabled
```

Enable it:

```code
$ tctl auth update-override --type=windows --public-key=<Var name="public-key" /> --set-disabled=false
```

You can find the public key of the target override by running
`tctl auth pub-key-hash --cert=cert.pem` or reading it from
`tctl get ca_overrides/windows`. Alternatively, you may edit directly using
`tctl edit ca_overrides/windows`.
</Admonition>

> ### Create certificate overrides on multi-Auth, HSM-enabled clusters

Reviewer: I'm not sure we use the phrase "multi-Auth" enough in the docs for users to immediately know what this means. Also, is there a more concise way to describe these clusters? "Shared HSM"? Since only the Auth Service manages CAs, there might be a way to not mention it in the heading.

Author: I think we typically call these HA (for High Availability), but even being HA doesn't mean the cluster will hit this corner-case. There isn't really a name for it, afaik.

"Shared HSM" would be the opposite, it's closer to "Exclusive HSM".

What I want to address here is an HA cluster (multiple Auths), where each Auth has an exclusive HSM instance (likely PKCS#11).

Author: An alternative:

* "Prepare CSRs for clusters with per-instance HSM keys"
* "Create certificate overrides on clusters with per-instance HSM keys"

"In clusters with per-instance HSM keys, run `tctl` locally against each Auth
Service instance. Each Auth Service instance generates CSRs for the keys it can
access."

It's a bit awkward, but it's not too long and is more to the point than "HSM enabled". There are many other variants we could try to.

WDYT?

Author: I went with the mouthful "multi-Auth clusters with per-instance HSM keys". (Happy to change it to something else.)

In multi-Auth, HSM-enabled clusters, each Auth Service instance may have
exclusive private keys. In this scenario, make sure all Auth instances are
online and healthy, then create your overrides in the disabled state. The Auth
instances automatically sign CRLs in the background for their respective keys.

> After a certificate override has a matching CRL it can be enabled.

Reviewer: ```suggestion
In multi-Auth, HSM-enabled clusters, each Auth Service instance may have
exclusive private keys. In this scenario, make sure all Auth Service instances are
online and healthy, then create your overrides in the disabled state. The Auth
Service instances automatically sign CRLs in the background for their respective keys.
After a certificate override has a matching CRL it can be enabled.
```
